src/components/TextField/TextField.jsx

```diff
@@ -176,15 +176,18 @@ const TextField = createClass({
 	},
 
 	handleBlur(event) {
-		const { onBlur } = this.props;
+		const { onBlur, onChangeDebounced } = this.props;
 
 		const value = _.get(event, 'target.value', '');
 
+		if (onChangeDebounced !== _.noop) {
+			this._handleChangeDebounced.flush();
```

Comment:
how about just calling flush always - instead of checking against `lodash.noop`...